src/hotspot/os/aix/os_aix.cpp

```diff
@@ -796,6 +796,8 @@ bool os::create_thread(Thread* thread, ThreadType thr_type,
   // OSThread::thread_id is the pthread id.
   osthread->set_thread_id(tid);
 
+  // child thread synchronization is not done here on AIX, a thread is started in suspended state
+
```

Comment:
It is interesting that on Windows, where we also start threads suspended, we also do:
```
// Thread state now is INITIALIZED, not SUSPENDED
  osthread->set_state(INITIALIZED);
```
then we have:
```
  // The thread is returned suspended (in state INITIALIZED), and is started higher up in the call chain
  return true;

```

Reply:
But I see now that states ALLOCATED and INITIALIZED are really only needed for the parent/child handshake.